File: Array/1004_max_consecutive_ones_III.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int left = 0, count = 0, max_count = 0;

        for(int right = 0; right < nums.size(); right += 1) {
            if(nums[right] == 0) {
                k -= 1;

                while(k < 0) {
                    if(nums[left] == 0) k += 1;
                    left += 1;
                    count -= 1;
                }
            }
            count += 1;
            max_count = max(max_count, count);
        }

        return max_count;
    }
};

File: Array/1004_max_consecutive_ones_III.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        longest, left = 0, 0

        for right in range(len(nums)):
            if nums[right] == 0:
                k -= 1

            while k < 0:
                if nums[left] == 0:
                    k += 1
                left += 1

            longest = max(longest, right - left + 1)

        return longest
    
# revised - 02/26/2025